packages/config-loader/src/schema/compile.ts

```diff
@@ -66,10 +77,9 @@ export function compileConfigSchemas(
             return false;
           }
           if (visibility && visibility !== 'backend') {
-            const normalizedPath = context.instancePath.replace(
-              /\['?(.*?)'?\]/g,
-              (_, segment) => `/${segment}`,
-            );
+            const normalizedPath = normalizeInstancePath(
+              context.instancePath,
+            ).replace(/\['?(.*?)'?\]/g, (_, segment) => `/${segment}`);
```

Comment:
I think we can bring the second replace into `normalizeInstancePath` too, feels like that's part of it

Reply:
Yup, makes sense. I also adjusted where we're using this to include error messages.